### PR TITLE
Support for Call Alias

### LINKED ISSUE #4: Support for call aliases
Ensure that when an invocation is requested, it supports the use of a call alias if that alias is stored in the cache.

### PR DESCRIPTION
**Pull request by brooksmtownsend**
Reopened issue from #48 

Fixes #4 

This PR also renames all `:ets` tables to "table", instead of "registry" to avoid confusion with the `Registry` module. 

This PR ALSO transforms the shutdown message publish into a `Gnats.req`, which allows the provider 2 seconds to clean up resources and respond. After the shorter length of time between 2 seconds and receiving a reply, the provider process is automatically terminated.

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
This appears to fail for the same reason as #55 , no unsubscribe logic for actors

File: host_core/lib/host_core/providers/provider_supervisor.ex [hunk 1]
```diff
@@ -34,7 +34,17 @@ defmodule HostCore.Providers.ProviderSupervisor do
     [{pid, _val}] = Registry.lookup(Registry.ProviderRegistry, {public_key, link_name})
     Logger.info("About to terminate child process")
     prefix = HostCore.Host.lattice_prefix()
-    :ok = Gnat.pub(:lattice_nats, "wasmbus.rpc.#{prefix}.#{public_key}.#{link_name}.shutdown", "")
+    # Allow provider 2 seconds to clean up resources
+    case Gnat.request(
```
Comment (autodidaddict) [inline host_core/lib/host_core/providers/provider_supervisor.ex:38]: Have the providers been modified to respond to this as a request? It used to be an async publish

**Review by autodidaddict — CHANGES_REQUESTED** [review]
Just would like to see the capability providers modified so they're responding to the shutdown request.

File: host_core/lib/host_core/providers/provider_supervisor.ex (hunk 1 above)
Comment (brooksmtownsend) [reply host_core/lib/host_core/providers/provider_supervisor.ex:38]: Just modified the two providers we have, and I added a Makefile so we can easily rebuild them. It's not a perfect makefile but it'll do for now, especially when I have to swap them out for the Linux executables for committing.

Feasibly, we could have that part automated as well.

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**